weaver/processes/wps_package.py

```diff
@@ -2114,18 +2145,47 @@ def _handler(self, request, response):
             self.update_status("Package complete.", PACKAGE_PROGRESS_DONE, STATUS_SUCCEEDED)
         return self.response
 
-    @staticmethod
-    def make_location_input(input_type, input_definition):
+    def must_fetch(self, input_ref):
+        # type: (AnyStr) -> bool
+        """
+        Figures out if file reference should be fetched immediately for local execution.
+        If anything else than local script/docker, remote ADES/WPS process will fetch it.
+        S3 are handled here to avoid error on remote WPS not supporting it.
+
+        .. seealso::
+            - :ref:`File Reference Types`
+        """
+        if self.is_ems or self.package_type == PROCESS_WORKFLOW:
+            return False
```

Comment:
Indeed there's different behavior happening here whether it is EMS/ADES. 
We could argue that one shouldn't call the process directly via ADES, and even more, remote WPS shouldn't be registered on an ADES in the first place, only on an EMS. 

So the correct behavior would be to forward references directly, which makes sense in my opinion.
We could have  a builtin process whose job is to do s3->https conversion. 
This would handle #183 quite nicely actually.